### PR TITLE
SHOWCASE-2327 - Change from NPM log to Winston log

### PR DESCRIPTION
**Pull request by kamorel**
# Description
Moving over from the npmlog/Morgan logging system and now we will be using JSON Format/Winston.

## Types of changes
Added Winston JSON Logging and changed all log calls to the new format.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments
Apologies for all the linting changes included by Prettier. I have disabled it going forward to avoid that.

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://getok-dev.apps.silver.devops.gov.bc.ca/pr-329

File: app/src/components/keyCloakServiceClientMgr.js [hunk 1]
```diff
@@ -72,63 +115,99 @@ class KeyCloakServiceClientManager {
...
       generatedPassword: clientSecret.value,
       generatedServiceClient: serviceClient.clientId,
-      oidcTokenUrl: this.svc.tokenUrl
+      oidcTokenUrl: this.svc.tokenUrl,
     };
   }
 
   async fetchClients(applicationAcronymList) {
-    log.debug('KeyCloakServiceClientManager.fetchClients', applicationAcronymList);
+    log.debug(applicationAcronymList, { function: 'fetchClients' });
```
Comment (jujaga) [inline app/src/components/keyCloakServiceClientMgr.js:169]:
```js
log.debug('applicationAcronymList', { function: 'fetchClients', applicationAcronymList: applicationAcronymList });
```

File: app/src/components/keyCloakServiceClientMgr.js [hunk 2]
```diff
@@ -137,49 +216,50 @@ class KeyCloakServiceClientManager {
...
       return undefined;
     }
   }
 
   // Get all users based on search param, supply undefined as searchParams to get ALL users in realm
   async findUsers(searchParams) {
-    log.debug('KeyCloakServiceClientManager.findUsers', `Params: ${JSON.stringify(searchParams)}`);
+    log.debug(`Params: ${JSON.stringify(searchParams)}`, {
+      function: 'findUsers',
+    });
```
Comment (jujaga) [inline app/src/components/keyCloakServiceClientMgr.js:240]:
```js
log.debug('Params', { function: 'findUsers', searchParams: searchParams });
```

File: app/src/components/keyCloakServiceClientMgr.js [hunk 3]
```diff
@@ -137,49 +216,50 @@ class KeyCloakServiceClientManager {
...
 
     const users = await this.svc.getUsers(searchParams);
 
     if (users && users.length) {
       return users;
     } else {
-      log.debug('KeyCloakServiceClientManager.findUsers', `No users found for ${JSON.stringify(searchParams)}`);
+      log.debug(`No users found for ${JSON.stringify(searchParams)}`, {
```
Comment (jujaga) [inline app/src/components/keyCloakServiceClientMgr.js:247]:
```js
log.debug('No users found', { function: 'findUsers', searchParams: searchParams });
```

File: app/src/components/permissionHelpers.js [hunk 4]
```diff
@@ -15,8 +15,13 @@ const permissionHelpers = {
...
     // Do they have access to the Acronym they are trying to POST
-    if (!acronyms.map(acr => acr.acronym).includes(applicationAcronym)) {
-      log.info('checkAcronymPermission', `User not authorized for acronym ${applicationAcronym}. User: ${JSON.stringify(userId)}`);
+    if (!acronyms.map((acr) => acr.acronym).includes(applicationAcronym)) {
+      log.info(
+        `User not authorized for acronym ${applicationAcronym}. User: ${JSON.stringify(
+          userId
+        )}`,
+        { function: 'checkAcronymPermission' }
+      );
```
Comment (jujaga) [inline app/src/components/permissionHelpers.js:24]:
```js
log.info(
  `User not authorized for acronym ${applicationAcronym}`,
  { function: 'checkAcronymPermission', userId: userId }
);
```

File: app/src/components/users.js [hunk 5]
```diff
@@ -46,36 +51,48 @@ const users = {
...
+      !acronymsFromDb.length
+    ) {
       return [];
     }
-    const acronyms = acronymsFromDb.map(acr => acr.acronym);
-    log.debug('getUserAcronymClients', `Acronyms for user: ${JSON.stringify(acronyms)}`);
+    const acronyms = acronymsFromDb.map((acr) => acr.acronym);
+    log.debug(`Acronyms for user: ${JSON.stringify(acronyms)}`, {
+      function: 'getUserAcronymClients',
+    });
```
Comment (jujaga) [inline app/src/components/users.js:69]:
```js
log.debug('Acronyms for user', { function: 'getUserAcronymClients', acronyms: acronyms });
```

File: app/src/components/utils.js [hunk 6]
```diff
@@ -37,21 +42,27 @@ const utils = {
...
+          auth: {
+            username: username,
+            password: password,
+          },
+          headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+        }
+      );
+      log.verbose(JSON.stringify(response.data), {
+        function: 'getKeyCloakToken',
       });
```
Comment (jujaga) [inline app/src/components/utils.js:61]:
```js
log.verbose('Keycloak response', { function: 'getKeyCloakToken', data: response.data });
```

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://getok-dev.apps.silver.devops.gov.bc.ca/pr-329

**Review by jujaga — APPROVED** [review]

**Pull request merged**